GenumKit/Stencil/CallNode.swift

```diff
@@ -0,0 +1,92 @@
+//
+// GenumKit
+// Copyright (c) 2016 Olivier Halligon
+// MIT Licence
+//
+
+import Stencil
+
+struct CallableBlock {
+  let parameters: [String]
+  let nodes: [NodeType]
+
+  func context(_ context: Context, arguments: [Variable]) throws -> [String: Any] {
+    var result = [String: Any]()
+
+    for (parameter, argument) in zip(parameters, arguments) {
+      result[parameter] = try argument.resolve(context)
+    }
+
+    return result
+  }
+}
+
+open class DefineNode: NodeType {
```

Comment:
Is it a convention in Stencil as well to make everything `open` like that (class, variables and the `parse` function) even for "leaf" (non-abstract) nodes?